_posts/2022-08-14-data-pipelines-production-ready-great-expectations.md

```diff
@@ -0,0 +1,104 @@
+---
+layout: post
+title: "Make Your Trino Data Pipelines Production Ready Using Great Expectations"
+author: "Brian Olsen, Brian Zhan"
+excerpt_separator: <!--more-->
+---
+
+One of the important aspects to writing good data pipelines is ensuring data
+quality by testing the data is what you're expecting it to be at any given
+state. Great Expectations is an open source Python library that allows you to 
+write detailed tests called
+[expectations](https://docs.greatexpectations.io/docs/terms/expectation/)
+against your data. Users write these expectations to run validations against the
+data as it enters your system. These expectations are expressed as methods in
+Python and stored in JSON and YAML files. One great advantage of expectations is
+the human readable documentation that results from these tests. As you roll out
+different versions of the code, you get alerted to any unexpected changes and 
+have version-specific generated documentation for what changed. In this blog, 
+we'll show you how to write expectations on tables in Trino!
+
+![](/assets/blog/data-pipelines-production-ready-great-expectations/trino-ge.png)
+
+
+<!--more-->
+
+## The need for data quality
+
+Managing data pipelines is not for the faint of heart. Nodes fail, you run
+out of memory, bursty traffic causes abnormal behavior, and that’s just the tip
+of the Apache Iceberg 😉. Lots of Trino community members build sophisticated
+data pipelines and data applications using Trino. Building data pipelines in
+Trino became more common with the addition of a
+[fault tolerance mode]({% post_url 2022-05-05-tardigrade-launch %}) to safeguard
+against some of these failures when executing long-running and 
+resource-intensive queries. 
+
+Aside from all the infrastructure problems that concern data teams, another
+category of problems that has been the silent killer for quite some time is data
+quality. Faulty data comes in, which can either cause data pipelines to fail, or
+it can possibly go unnoticed and cause inaccurate downstream reporting. Tacit 
+knowledge is scattered among domain experts, technical experts, and the code and
```

Comment:
```suggestion
it can possibly go unnoticed and cause inaccurate downstream reporting. 
Knowledge is scattered among domain experts, technical experts, and the code and
```